src/trans/gpu/internal/trgtol_mod.F90

```diff
@@ -1243,67 +1260,102 @@ SUBROUTINE TRGTOL(PGLAT,KF_FS,KF_GP,KF_SCALARS_G,KVSET,KPTRGP,&
     ENDDO
   
     IPOS=0
+    JK_MAX = 0
     DO JBLK=1,NGPBLKS
       IGPTROFF(JBLK)=IPOS
       IFIRST = IGPTRSEND(1,JBLK,MYSETW)
       IF(IFIRST > 0) THEN
         ILAST = IGPTRSEND(2,JBLK,MYSETW)
         IPOS=IPOS+ILAST-IFIRST+1
+        IF (JK_MAX<(ILAST-IFIRST+1)) JK_MAX = (ILAST-IFIRST+1)
       ENDIF
     ENDDO
-
+#ifdef ACCGPU
+    !$ACC DATA COPYIN(IFLDOFF,IGPTROFF,LLGP2,LLGP3A,LLGP3B,LLUV,IGPTRSEND)
+#endif
+#ifdef OMPGPU
+    !$OMP TARGET DATA MAP(TO:IFLDOFF,IGPTROFF,LLGP2,LLGP3A,LLGP3B,LLUV,IGPTRSEND)
+#endif
     CALL GSTATS(1601,0)
-#ifdef NECSX
-  !$OMP PARALLEL DO SCHEDULE(DYNAMIC) PRIVATE(JFLD,JBLK,JK,IFLD,IPOS,IFIRST,ILAST)
-#else
-  !$OMP PARALLEL DO SCHEDULE(STATIC) PRIVATE(JFLD,JBLK,JK,IFLD,IPOS,IFIRST,ILAST)
+    IF(LLPGPONLY) THEN
+#ifdef ACCGPU
+      !$ACC PARALLEL LOOP COLLAPSE(3) PRIVATE(IPOS,IFIRST,ILAST,IFLD,JK)
```

Comment:
@reuterbal - we are aiming for all parallel regions to have `DEFAULT(NONE)` right? In which case this would have to be added to all new parallel regions in this file. `PGLAT` would have to be explicitly declared then as well (and maybe other variables I missed).

Reply:
I can comply with style.

Reply:
I've just updated the PR which complies, mostly, with the style conventions.

Reply:
I think it's because `IGP3BLEVS` is only conditionally present. If `PRESENT(PGP3B) == .FALSE.` then it will not be copied to device on [line 1244](https://github.com/PaulMullowney/ectrans/blob/ftdir_opt/src/trans/gpu/internal/trgtol_mod.F90#L1244C18-L1244C32) and hence won't be in the present table for the enclosed compute constructs. Talking to @reuterbal about this, we think that you can delete `IGP3BLEVS` and other variables from the present lists of all compute constructs inside the main data region (starting [line 1237](https://github.com/PaulMullowney/ectrans/blob/ftdir_opt/src/trans/gpu/internal/trgtol_mod.F90#L1237)). Even though the compute constructs have `DEFAULT(NONE)`, we think it shouldn't be a problem because they inherit the present list of the enclosing data region. The OpenACC standard is not entirely clear on this though. For now I suggest to delete `IGP3BLEVS` and other problematic variables from the present list.